### PR TITLE
[Add] add optional selection for filter order

### PR DESCRIPTION
**Pull request by roganw**
Add a chance to change the order of optional value. 
When left empty, it will order the optional value by metric.

### CODE REVIEW

**Comment by mistercrunch** [comment]
It's a bit better than before, but ultimately we'll want for each filter to be able to specify its own metric (or no metric at all) and its own ordering preference.

**Comment by RoganW** [comment]
@mistercrunch Yeah, I think your proposal is more reasonable.
I'll optimize that later.

**Comment by RoganW** [comment]
@mistercrunch Already optimized

**Comment by mistercrunch** [comment]
What I'm proposing would be much more complicated. We may want to just land this as it is in the meantime.

File: superset/assets/backendSync.json [hunk 1]
```diff
@@ -2006,6 +2006,13 @@
       "default": false,
       "description": "Whether to include a time filter"
     },
+    "filter_order": {
+      "type": "SelectControl",
+      "multi": true,
+      "label": "Order by optional value",
```
Comment (mistercrunch) [inline superset/assets/backendSync.json]: `"Ordering"`

File: superset/assets/javascripts/explore/stores/controls.jsx [hunk 2]
```diff
@@ -955,6 +955,18 @@ export const controls = {
     description: 'Whether to include a time filter',
   },
 
+  filter_order: {
+    type: 'SelectControl',
+    multi: true,
```
Comment (mistercrunch) [inline superset/assets/javascripts/explore/stores/controls.jsx]: shouldn't this be false?

File: superset/assets/javascripts/explore/stores/controls.jsx [hunk 3]
```diff
@@ -955,6 +955,18 @@ export const controls = {
...
+    label: 'Order by optional value',
+    default: [],
+    mapStateToProps: state => ({
+      choices: (state.datasource) ? state.datasource.order_by_choices : [],
```
Comment (mistercrunch) [inline superset/assets/javascripts/explore/stores/controls.jsx]: shouldn't the option be `alphabetically`, `by metric descending`, and `by metric ascending`?

File: superset/assets/javascripts/explore/stores/controls.jsx [hunk 4]
```diff
@@ -955,6 +955,18 @@ export const controls = {
...
+    }),
+    description: 'The way to order tho optional value of filters. ' +
+    'When does\'t select for one filter, it will order the optional value by metric.',
```
Comment (mistercrunch) [inline superset/assets/javascripts/explore/stores/controls.jsx]: use `"` when string contains apostrophe `'`

File: superset/assets/javascripts/explore/stores/controls.jsx [hunk 5]
```diff
@@ -955,6 +955,18 @@ export const controls = {
...
 
+  filter_order: {
+    type: 'SelectControl',
+    multi: true,
+    label: 'Order by optional value',
+    default: [],
+    mapStateToProps: state => ({
+      choices: (state.datasource) ? state.datasource.order_by_choices : [],
+    }),
+    description: 'The way to order tho optional value of filters. ' +
```
Comment (mistercrunch) [inline superset/assets/javascripts/explore/stores/controls.jsx]: tho -> the

Comment (mistercrunch) [inline superset/assets/javascripts/explore/stores/controls.jsx]: "How to order the list of values for the filter"

**Review by mistercrunch — CHANGES_REQUESTED** [review]

File: superset/assets/javascripts/explore/stores/controls.jsx (hunk 2 above)
Comment (RoganW) [inline superset/assets/javascripts/explore/stores/controls.jsx]: Via setting `true`, we can specify ordering preference for each filter.

File: superset/assets/javascripts/explore/stores/controls.jsx (hunk 3 above)
Comment (RoganW) [inline superset/assets/javascripts/explore/stores/controls.jsx]: I think, in most cases, users know which option they want to choose, and we just need to help them to find it quickly. so, be `alphabetically` and be `reverse order` are used more often than `by metric descending`. 
For example, area(alphabetically) and period(like year-month in reverse order) are used regularly in sales analysis.

Comment (RoganW) [inline superset/assets/javascripts/explore/stores/controls.jsx]: If I want to compare the options ordering by`metric descending` or `metric ascending`, I  would prefer to create a new slice specially to show that.

File: superset/assets/javascripts/explore/stores/controls.jsx (hunk 2 above)
Comment (mistercrunch) [inline superset/assets/javascripts/explore/stores/controls.jsx]: but don't you run out of options then? each option can only be used once isn't it?

Comment (mistercrunch) [inline superset/assets/javascripts/explore/stores/controls.jsx]: oh I see, but what if you want two different filters to be sorted on the same metric?

Comment (RoganW) [inline superset/assets/javascripts/explore/stores/controls.jsx]: By default, filters will be sorted by metric in descending order, if not select any`filter_order ` related to them.

**Comment by mistercrunch** [comment]
Notice: this issue has been closed because it has been inactive for 250 days. Feel free to comment and request for this issue to be reopened.

**Pull request closed without merging**